File: youtubejextractor/src/main/java/com/github/kotvertolet/youtubejextractor/models/test/ButtonRenderer.java
package com.github.kotvertolet.youtubejextractor.models.test;

import com.google.gson.annotations.SerializedName;

public class ButtonRenderer{

	@SerializedName("trackingParams")
	private String trackingParams;

	@SerializedName("size")
	private String size;

	@SerializedName("style")
	private String style;

	@SerializedName("text")
	private Text text;

	@SerializedName("serviceEndpoint")
	private ServiceEndpoint serviceEndpoint;

	@SerializedName("navigationEndpoint")
	private NavigationEndpoint navigationEndpoint;

	public void setTrackingParams(String trackingParams){
		this.trackingParams = trackingParams;
	}

	public String getTrackingParams(){
		return trackingParams;
	}

	public void setSize(String size){
		this.size = size;
	}

	public String getSize(){
		return size;
	}

	public void setStyle(String style){
		this.style = style;
	}

	public String getStyle(){
		return style;
	}

	public void setText(Text text){
		this.text = text;
	}

	public Text getText(){
		return text;
	}

	public void setServiceEndpoint(ServiceEndpoint serviceEndpoint){
		this.serviceEndpoint = serviceEndpoint;
	}

	public ServiceEndpoint getServiceEndpoint(){
		return serviceEndpoint;
	}

	public void setNavigationEndpoint(NavigationEndpoint navigationEndpoint){
		this.navigationEndpoint = navigationEndpoint;
	}

	public NavigationEndpoint getNavigationEndpoint(){
		return navigationEndpoint;
	}
}

File: youtubejextractor/src/main/java/com/github/kotvertolet/youtubejextractor/models/newModels/ButtonRenderer.kt
package com.github.kotvertolet.youtubejextractor.models.newModels

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import java.io.Serializable

@Parcelize
class ButtonRenderer : Parcelable, Serializable {
    @SerializedName("trackingParams")
    var trackingParams: String? = null

    @SerializedName("size")
    var size: String? = null

    @SerializedName("style")
    var style: String? = null

    @SerializedName("text")
    var text: Text? = null

    @SerializedName("serviceEndpoint")
    var serviceEndpoint: ServiceEndpoint? = null

    @SerializedName("navigationEndpoint")
    var navigationEndpoint: NavigationEndpoint? = null
    override fun toString(): String {
        return "ButtonRenderer{" +
                "trackingParams = '" + trackingParams + '\'' +
                ",size = '" + size + '\'' +
                ",style = '" + style + '\'' +
                ",text = '" + text + '\'' +
                ",serviceEndpoint = '" + serviceEndpoint + '\'' +
                ",navigationEndpoint = '" + navigationEndpoint + '\'' +
                "}"
    }
}